app/models/concerns/waste_carriers_engine/can_have_registration_attributes.rb

```diff
@@ -23,11 +22,30 @@ module CanHaveRegistrationAttributes
       # rubocop:disable Layout/LineLength
       embeds_many :addresses, class_name: "WasteCarriersEngine::Address"
 
-      # This is our own custom association. See CanReferenceSingleDocumentInCollection for details
-      reference_one :contact_address, collection: :addresses, find_by: { address_type: "POSTAL" }
-      reference_one :company_address, collection: :addresses, find_by: { address_type: "REGISTERED" }
-      # TODO: Remove this and always use `company_address` rather than `registrered_address`
-      reference_one :registered_address, collection: :addresses, find_by: { address_type: "REGISTERED" }
+      # Define helper accessor and assignment methods for each address type
+      %w[contact postal registered].each do |address_type|
+
+        define_method(:"#{address_type}_address") do
+          # handle synonyms
+          address_type = "postal" if address_type == "contact"
+
+          addresses.where(address_type: address_type.upcase).first
+        end
+
+        define_method(:"#{address_type}_address=") do |address|
+          # handle synonyms
+          address_type = "postal" if address_type == "contact"
+
+          unless address.blank? || address.address_type == address_type.upcase
+            raise ArgumentError, "Attempted to set #{address_type} address with address of type #{address.address_type}"
+          end
+
+          # clear any existing address of this type
+          addresses.where(address_type: address_type.upcase).first&.destroy!
+
+          addresses << address
+        end
+      end
```

Comment:
I don't think this matches what was there before.
Previously, we had 

contact_address -> address_type: POSTAL
company_address -> address_type: REGISTERED
registered_address -> address_type: REGISTERED

Now we have

contact_address -> address_type: POSTAL
postal_address -> address_type: POSTAL
registered_address -> address_type: REGISTERED

Please double check if code is correct

Reply:
I agree postal_address is redundant. The rest looks correct - company_address is no more.